environment.yml

```diff
@@ -2,18 +2,8 @@ name: pgeon-dev
 channels:
   - conda-forge
 dependencies:
-  # required
-  - python
-  - arrow-cpp
-  - pyarrow
-  - libpq
-
-  # The compiler packages are meta-packages and install the correct compiler (activation) packages on the respective platforms.
-  - c-compiler
   - cxx-compiler
-
-  # Dev
   - git
-  - cmake
-  - ninja
-  - postgresql
+  - libpq
+  - pyarrow
```

Comment:
Isn't this getting pulled in the environment?

Reply:
I'm not sure what you mean. These are the minimal req for building and testing the python package in the dev container. I'd like to be able to remove pyarrow but as discussed this does not work at the moment in dev container (but works fine on macos)

Reply:
The reason why it works on mac, is because mac is somehow a bit backward compatible and it doesn't have the problem of distribution compatibility on Linux (and the need of `manylinux`).

This `pyarrow` you're listing here is built "wrong", i.e. it uses the cxx-compiler from conda and not the `manylinux` environment. That is why you're getting the missing symbol, I think.

You need to build the package on `manylinux`, and also `arrow` and `pyarrow` need to be built on `manylinux`, or, from what I understand, it won't work correctly.

Reply:
You're getting it reverse I think.
At the moment with conda pyarrow installed we can pip install . and run pgeon both on macos and in the dev container

If I remove the pyarrow conda install (and arrow-cpp) and just let pip deal with pyarrow, it works on macos but not on linux.

I just tried to use a simple Dockerfile with no conda at all
```
FROM python:3
```

Then in the container, simply
```
CPPFLAGS="-I/usr/include/postgresql" pip install .
```

It builds but then it does not run (it doesn't even matter whether or not the build is isolated)
```
ImportError: /usr/local/lib/python3.11/site-packages/_pgeon.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZNK5arrow8DataType18ComputeFingerprintB5cxx11Ev
```

There's no conda at all